spec/code_block_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'spec_helper'
+require 'kramdown'
+require 'ap'
+require 'pp'
+site = File.join(File.dirname(__FILE__), '..','content', '**', '*.md')
+PAGES = Dir.glob(site)
+
+def extract_code(element,prev_element)
+  codes = [] 
+  if element.type == :codespan \
+    && !element.value.nil? \
+    && !codes.include?(element.value) \
+    && ((prev_element.children.size == 1) || (element.value[/^\n/])) #element value hasn't got any sibling or begin with '\n' 
+      codes.push element.value
+  end
+  element.children.each do |child|
+    codes += extract_code(child, element)
+  end
+  codes
+end
+
+PAGES.each do |page|
+  generated_page = Kramdown::Document.new(File.read(page))
+  codes = extract_code(generated_page.root,generated_page)
```

Comment:
It seems like you could pass just `generated_page` instead and use `.root` where needed.